src/EditorFeatures/CSharp/Highlighting/KeywordHighlighters/SwitchStatementHighlighter.cs

```diff
@@ -41,35 +44,39 @@ internal class SwitchStatementHighlighter : AbstractKeywordHighlighter<SwitchSta
         /// Finds all breaks and continues that are a child of this node, and adds the appropriate spans to the spans
         /// list.
         /// </summary>
-        private void HighlightRelatedKeywords(SyntaxNode node, List<TextSpan> spans)
+        private void HighlightRelatedKeywords(SyntaxNode node, List<TextSpan> spans, bool highlightBreaks, bool highlightGotos)
         {
-            switch (node)
-            {
-                case BreakStatementSyntax breakStatement:
-                    spans.Add(breakStatement.BreakKeyword.Span);
-                    spans.Add(EmptySpan(breakStatement.SemicolonToken.Span.End));
-                    break;
+            Debug.Assert(highlightBreaks || highlightGotos);
 
-                case GotoStatementSyntax gotoStatement:
-                    var start = gotoStatement.GotoKeyword.SpanStart;
-                    var end = !gotoStatement.CaseOrDefaultKeyword.IsKind(SyntaxKind.None)
-                        ? gotoStatement.CaseOrDefaultKeyword.Span.End
-                        : gotoStatement.GotoKeyword.Span.End;
+            if (highlightBreaks && node is BreakStatementSyntax breakStatement)
```

Comment:
I thought about keeping this as a switch with highlightBreaks inside `when` (and doing the same in LoopHighlighter).. not sure if that would be the best thing to do here but at least i'd be dogfooding my own highlighting :D